fuzz/fuzz_targets/pandoc.rs

```diff
@@ -0,0 +1,170 @@
+#![no_main]
+
+//! Differential fuzzing of pulldown-cmark and pandoc commonmark.
+//!
+//! This fuzzer sends the same input to both Markdown parsers and
+//! compares the output. Pandoc's JSON AST is turned into
+//! `pulldown_cmark::Event` values for this purpose.
+//!
+//! Run the fuzzer like this to only test ASCII input (which is
+//! usually enough to find parsing differences):
+//!
+//!     cargo fuzz run pandoc -- -only_ascii=1
+
+use libfuzzer_sys::fuzz_target;
+use pretty_assertions::assert_eq;
+use pulldown_cmark_fuzz::{PandocHandle, normalize_pandoc, print_events, pulldown_cmark_ext, xml_to_events, commonmark_js};
+use std::sync::OnceLock;
+
+static PANDOC: OnceLock<PandocHandle> = OnceLock::new();
+
+fuzz_target!(|text: String| {
+    // There are some differences in handling of non-UTF-8 input.
+    if text.bytes().any(|b| b.is_ascii_control() && b != b'\n') {
+        return;
+    }
+
+    // https://github.com/raphlinus/pulldown-cmark/issues/659
+    if text.contains("<") {
+        return;
+    }
+
+    // https://github.com/raphlinus/pulldown-cmark/issues/661
+    if text.contains("`") {
+        return;
+    }
+
+    if text.bytes().any(|b| b > 127) {
+        return;
+    }
+
+    // There are some trivial differences due to trailing whitespace.
+    let mut text = text
+        .lines()
+        .map(|line| line.trim_end())
+        .collect::<Vec<_>>()
+        .join("\n");
+    text.push('\n');
+
+    let pulldown_cmark_events = pulldown_cmark_ext(&text);
+
+    // Make sure there aren't cyclical footnotes.
+    // pulldown-cmark supports them, but pandoc does not
+    // and making it work would require completely redesigning
+    // the pandoc ast.
+    //
+    // commonmark-hs also trims footnote reference names.
+    // I'm not going to bother reporting it as a bug, since it's
+    // obviously on purpose and it's not gonna matter.
+    let mut footstack = vec![];
+    for event in &pulldown_cmark_events {
+        use pulldown_cmark::{Event, Tag, TagEnd};
+        match event {
+            Event::Start(Tag::FootnoteDefinition(id)) => {
+                if id.starts_with("\n") || id.ends_with("\n") || id.starts_with("\r") || id.ends_with("\r") || id.starts_with(" ") || id.starts_with("\t") || id.contains("  ") || id.contains("\t ") || id.contains(" \t") || id.contains("\t\t") || id.ends_with(" ") || id.ends_with("\t") { return };
```

Comment:
Perhaps it would be simpler to use the slice variants of the `starts_with` and `ends_with` patterns:

```suggestion
                let whitespace = &['\n', '\r', ' ', '\t'];
                if id.starts_with(whitespace)
                    || id.ends_with(whitespace)
                    || id.contains("\t ")
                    || id.contains(" \t")
                    || id.contains("\t\t")
                {
                    return;
                };
```